### PR TITLE
settings: ban cockroachdb imports

### PR DESCRIPTION
**Pull request by dt**

### CODE REVIEW

File: build/style_test.go [hunk 1]
```diff
@@ -502,12 +502,13 @@ func TestStyle(t *testing.T) {
 			}
 			return nil
 		})
+		settingsPkg := `github.com/cockroachdb/cockroach/pkg/settings`
```
Comment (tamird) [inline build/style_test.go]: looks like the reusable bit is really `settingsPkg+": "`, maybe make that the extracted string and call it settingsPkgImports or some such

File: build/style_test.go [hunk 2]
```diff
@@ -527,6 +528,10 @@ func TestStyle(t *testing.T) {
 				t.Errorf(`%s <- please use "golang.org/x/net/context" instead of "context"`, s)
 			case strings.HasSuffix(s, " syscall"):
 				t.Errorf(`%s <- please use "golang.org/x/sys" instead of "syscall"`, s)
+			case strings.HasPrefix(s, settingsPkg+": github.com/cockroachdb/cockroach"):
+				if !strings.HasSuffix(s, "testutils") && !strings.HasSuffix(s, "syncutil") {
+					t.Errorf("%s <- please don't add dependencies to settings pkg", s)
```
Comment (tamird) [inline build/style_test.go]: CRDB dependencies

**Review by tamird — APPROVED** [review]

**Comment by dt** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[build/style_test.go, line 505 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15201#-KiAdXGJIIkcwRgtT9ec-r1-505:-KiAeuHXIC27PEb4_idO:bhdc2cw) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e974eb1a05cfa91d0134250ad4f81adf005156b/build/style_test.go#L505)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

looks like the reusable bit is really `settingsPkg+": "`, maybe make that the extracted string and call it settingsPkgImports or some such
</blockquote></details>

heh, i started with that and then thought it was slightly clearer punctuation at the usage site.
Done.

---

*[build/style_test.go, line 533 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15201#-KiAdXGJIIkcwRgtT9ec-r1-533:-KiAfJNKN9ELgzX_Va_C:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e974eb1a05cfa91d0134250ad4f81adf005156b/build/style_test.go#L533)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

CRDB dependencies
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15201)*

**Comment by knz** [comment]
LGTM

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15201)*

**Comment by tamird** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15201#-:-KiAjtNzH5ICaCq4YLyg:bnfp4nl)*

**Pull request merged**